File: AquariumApi.Models/CameraConfiguration.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;
using System.Text;

namespace AquariumApi.Models
{
    public class CameraConfiguration
    {
        public int? Id { get; set; }
        public int Height { get; set; } = 1080;
        public int Width { get; set; } = 1920;
        public int Sharpness { get; set; } = 0;
        public int Contrast { get; set; } = 0;
        public int Saturation { get; set; } = 0;
        public string ExposureMode { get; set; } = "auto";
        public int Brightness { get; set; } = 0;
        public int Iso { get; set; } = 100;
        public bool HFlip { get; set; } = false;
        public bool VFlip { get; set; } = false;
        public decimal RoiX { get; set; } = 0;
        public decimal RoiY { get; set; } = 0;
        public decimal RoiW { get; set; } = 0;
        public decimal RoiH { get; set; } = 0;
        public int Rotation { get; set; } = 180;

        public string Output;

        public override string ToString()
        {
            return $"-o '{Output}' " +
                $"-h {Height} " +
                $"-w {Width} " +
                $"-sh '{Sharpness}' " +
                $"-co '{Contrast}' " +
                $"-sa '{Saturation}' " +
                $"-br '{Brightness}' " +
                $"-ISO {Iso} " +
                $"-ev 0 " +
                $"-ex '{ExposureMode}' " +
                $"{(HFlip ? "-hf": "")} " +
                $"{(VFlip ? "-vf":"")} " +
                $"-rot {Rotation} " +
                $"-roi {RoiX},{RoiY},{RoiW},{RoiH} ";
        }
    }
}

File: AquariumApi.DeviceApi/ClientApp/src/app/models/CameraConfiguration.ts
class Vector {
    constructor(public x, public y, public z) { }
}
export class CameraConfiguration {
    id: number;
    height = 1080;
    width = 1920;
    sharpness = 0;
    contrast = 0;
    saturation = 0;
    brightness = 50;
    exposureMode = 'auto';
    iso = 100;
    hFlip = false;
    vFlip = false;
    roiX = 0;
    roiY = 0;
    roiW = 1;
    roiH = 1;
    rotation = 0;
}
export const CameraExposureModes = [
    { value: "auto", viewValue: "auto" },
    { value: "night", viewValue: "night" },
    { value: "nightpreview", viewValue: "nightpreview" },
    { value: "backlight", viewValue: "backlight" },
    { value: "spotlight", viewValue: "spotlight" },
    { value: "sports", viewValue: "sports" },
    { value: "beach", viewValue: "beach" },
    { value: "verylong", viewValue: "verylong" },
    { value: "fixedfps", viewValue: "fixedfps" },
    { value: "antishake", viewValue: "antishake" },
    { value: "fireworks", viewValue: "fireworks"}
];